src/model/results-check.ts

```diff
@@ -22,9 +22,18 @@ const ResultsCheck = {
       files.map(async filepath => {
         if (!filepath.endsWith('.xml')) return;
         core.info(`Processing file ${filepath}...`);
-        const fileData = await ResultsParser.parseResults(path.join(artifactsPath, filepath));
-        core.info(fileData.summary);
-        runs.push(fileData);
+        try {
+          const content = fs.readFileSync(path.join(artifactsPath, filepath), 'utf8');
+          if (!content.includes('<test-run')) {
+            // noinspection ExceptionCaughtLocallyJS
+            throw new Error('File does not appear to be a NUnit XML file');
+          }
+          const fileData = await ResultsParser.parseResults(path.join(artifactsPath, filepath));
+          core.info(fileData.summary);
+          runs.push(fileData);
+        } catch (error: any) {
+          core.warning(`Failed to parse ${filepath}: ${error.message}`);
+        }
```

Comment:
We could probably go without the throw and just return after logging, right?

```suggestion
        const fullFilePath = path.join(artifactsPath, filepath)
        const content = fs.readFileSync(fullFilePath, 'utf8');
        
        if (!content.includes('<test-run')) {
          core.warning(`Failed to parse ${filePath}: File does not appear to be a NUnit XML file`);
          return
        }

        const fileData = await ResultsParser.parseResults(fullFilePath);
        core.info(fileData.summary);
        runs.push(fileData);
```

Reply:
Return statement proposed in https://github.com/game-ci/unity-test-runner/issues/288